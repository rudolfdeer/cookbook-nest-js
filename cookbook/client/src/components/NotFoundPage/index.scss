@import '../../styles/index.scss';

.page--not-found {
  position: relative;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;

  &__bg {
    width: 607px;
    height: 858px;
    position: absolute;
    z-index: -1;
    left: 50%;
  }

  &__content {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    overflow: hidden;
  }

  &__title {
    @include title-40;
    font-size: 70px;

    &--small {
     @include title-40; 
    }
  }

  &__text {
    @include text-18;

    a {
      @include text-18;
      color: $primary-yellow;
      text-decoration: underline;
    }
  }
}
