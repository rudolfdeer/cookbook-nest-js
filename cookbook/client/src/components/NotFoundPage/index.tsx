import React from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import ROUTES from '../../constants/routes';

import './index.scss';

export default function NotFoundPage(): JSX.Element {
  const { t } = useTranslation();

  return (
    <main className="page--not-found">
      <img
        src="../../../assets/images/pear-bg.png"
        className="page--not-found__bg"
      />
      <div className="page--not-found__content">
        <h1 className="page--not-found__title">404</h1>
        <h2 className="page--not-found__title--small">{t('NOT_FOUND')}</h2>
        <p className="page--not-found__text">
          {t('PAGE_NOT_EXIST')} <Link to={ROUTES.HOME}>{t('HOME_PAGE')}</Link>
        </p>
      </div>
    </main>
  );
}
