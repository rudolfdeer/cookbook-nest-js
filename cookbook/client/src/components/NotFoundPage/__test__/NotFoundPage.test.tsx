import React from 'react';
import { shallow } from 'enzyme';
import NotFoundPage from '../index';

describe('not found page component', () => {
  const component = shallow(<NotFoundPage />);
  it('renders the <NotFoundPage /> component', () => {
    expect(component).toMatchSnapshot();
  });
});
