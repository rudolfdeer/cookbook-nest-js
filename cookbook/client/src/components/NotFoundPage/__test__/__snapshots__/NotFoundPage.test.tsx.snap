// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not found page component renders the <NotFoundPage /> component 1`] = `ShallowWrapper {}`;
